monix-catnap-micrometer/src/main/scala/com/avast/sst/monix/catnap/micrometer/MicrometerCircuitBreakerMetrics.scala

```diff
@@ -0,0 +1,39 @@
+package com.avast.sst.monix.catnap.micrometer
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import cats.effect.Sync
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State.{Closed, HalfOpen, Open}
+import io.micrometer.core.instrument.MeterRegistry
+
+object MicrometerCircuitBreakerMetrics {
+
+  private[micrometer] class Impl[F[_]: Sync](name: String, meterRegistry: MeterRegistry) extends CircuitBreakerMetrics[F] {
+
+    private val F = Sync[F]
+
+    private[this] val CircuitOpened = -1
+    private[this] val CircuitHalfOpened = 0
+    private[this] val CircuitClosed = 1
+
+    private val accepted = meterRegistry.counter(s"circuit-breaker.$name.accepted")
+    private val rejected = meterRegistry.counter(s"circuit-breaker.$name.rejected")
+    private val circuitState = meterRegistry.gauge[AtomicInteger](s"circuit-breaker.$name.state", new AtomicInteger(CircuitClosed))
```

Comment:
i think MVar would be more idimoatic for this

Reply:
don't you mean `IORef`, or whatever it's called (`MVar`s can be empty, which doesn't make much sense for `circuitState`).
On one hand I would agree, it would be a nice pure FP approach.
On the other hand, it's totally private internal and I would expect `IORef` comes with more overhead than just `AtomicInteger`, and I don't know if it's worth it
:man_shrugging:

Reply:
after a bit of thinking, I would be for the FP approach, the overhead shouldn't be any (or be significant), because it's essentially doing the same thing, and using `IORef` with proper enum communicates the _intent_ better, then the lowlevel mechanics with `AtomicInteger`.

Reply:
Gauge is defined as this:

```java
public <T extends Number> T gauge(String name, T number)
```

It requires `T` to be `Number` and `AtomicInteger` satisfies that. There would be a possibility to use another method and use `Ref` (or whatever) but I don't think it's worth the trouble (I would have to provide a method to convert the reference to `Number` which means I would have to run the effect) considering it is totally internal implementation.